pkg/rpc/client/rpc.go

```diff
@@ -513,27 +512,23 @@ func (c *Client) CalculateValidUntilBlock() (uint32, error) {
 }
 
 // AddNetworkFee adds network fee for each witness script and optional extra
-// network fee to transaction.
-func (c *Client) AddNetworkFee(tx *transaction.Transaction, extraFee int64, acc *wallet.Account) error {
+// network fee to transaction. Scripts is an array verification signer's verification scripts.
+// nil means that signer is a contract, otherwise it must be either simple or multisig script.
+func (c *Client) AddNetworkFee(tx *transaction.Transaction, extraFee int64, scripts ...[]byte) error {
 	size := io.GetVarSize(tx)
-	if acc.Contract != nil {
-		netFee, sizeDelta := core.CalculateNetworkFee(acc.Contract.Script)
-		tx.NetworkFee += netFee
-		size += sizeDelta
-	}
-	for _, cosigner := range tx.Signers {
-		script := acc.Contract.Script
-		if !cosigner.Account.Equals(hash.Hash160(acc.Contract.Script)) {
+	for i, cosigner := range tx.Signers {
+		if scripts[i] == nil {
```

Comment:
Scripts length might be less than the length of tx.Signers.

Reply:
Fixed, added test.